src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java

```diff
@@ -0,0 +1,688 @@
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Entity> getEntities(long parentId, long entityTypeId) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<Entity> query = session.createQuery("FROM EntityImpl entity "
+				+ "WHERE entity.typeId = :entityTypeId AND entity.parentId = :parentId",
+				Entity.class)
+				.setParameter("entityTypeId", entityTypeId)
+				.setParameter("parentId", parentId);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public EntityType getEntityType(long id) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<EntityType> query = session.createQuery("FROM EntityTypeImpl entityType WHERE entityType.id = :id",
+				EntityType.class)
+				.setParameter("id", id);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::getSingleResult, null);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<EntityType> getEntityTypes() {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<EntityType> query = session.createQuery("FROM EntityTypeImpl entityType",
+				EntityType.class);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public int getEntityTypesCount() {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Table entityTypeTable = Entity.class.getAnnotation(Table.class);
+		String entityTypeTableName = "EntityType";
+
+		if (entityTypeTable != null) {
+			entityTypeTableName = entityTypeTable.name();
+		}
+
+		String sqlQuery = String.format("SELECT count(*) FROM %1$s", entityTypeTableName);
+		Query query = session.createNativeQuery(sqlQuery);
+
+		String stringResult = databaseAccess.getArtifacts(session, query, transaction, query::getSingleResult, 0).toString();
+		return Integer.parseInt(stringResult);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Event> getEvents(long eventTypeId, int listStartIndex, int listEndIndex, Long... entityIds) {
+		if (entityIds.length == 0) {
+			return new ArrayList<>();
+		}
+
+		StringBuilder sqlWhere = new StringBuilder();
+		sqlWhere.append(String.format("(event.entityId = %1$d)", entityIds[0]));
+
+		for (int i = 1; i < entityIds.length; i++) {
+			sqlWhere.append(String.format(" OR (event.entityId = %1$d)", entityIds[i]));
+		}
+
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		String sqlQuery = String.format(
+				"FROM EventImpl event "
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "FROM EventImpl event " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)